rust/core-lib/src/tabs.rs

```diff
@@ -735,16 +737,20 @@ impl CoreState {
 
     /// Handles a config related file system event.
     #[cfg(feature = "notify")]
-    fn handle_config_fs_event(&mut self, event: DebouncedEvent) {
-        use self::DebouncedEvent::*;
-        match event {
-            Create(ref path) | Write(ref path) | Chmod(ref path) => {
-                self.load_file_based_config(path)
+    fn handle_config_fs_event(&mut self, event: Event) {
+        use notify::event::*;
+        match event.kind {
+            EventKind::Create(CreateKind::Any)
+            | EventKind::Modify(ModifyKind::Any)
+            | EventKind::Modify(ModifyKind::Metadata(MetadataKind::Any)) => {
+                self.load_file_based_config(&event.paths[0])
             }
-            Remove(ref path) if !path.exists() => self.remove_config_at_path(path),
-            Rename(ref old, ref new) => {
-                self.remove_config_at_path(old);
-                self.load_file_based_config(new);
+            EventKind::Remove(RemoveKind::Any) if !event.paths[0].exists() => {
+                self.remove_config_at_path(&event.paths[0])
+            }
+            EventKind::Modify(ModifyKind::Name(RenameMode::Both)) => {
+                self.remove_config_at_path(&event.paths[0]);
+                self.load_file_based_config(&event.paths[1]);
```

Comment:
Implemented this.

Reply:
I agree I'd like to see bounds checks before we do this indexing. @sjoshid did you maybe forget to push your change?

Reply:
Bump this comment; this will crash if one of the paths is missing, which I believe is possible; for instance if the file was moved somewhere that we don't have permission to access.

Reply:
Agree. But that will never happen. `wants_event` will catch such invalid events.